src/nupic/algorithms/KNNClassifier.py

```diff
@@ -1020,6 +1067,12 @@ def getAdaptiveSVDDims(self, singularValues, fractionOfMax=0.001):
 
 
   def finalizeSVD(self, numSVDDims=None):
+    """
+    TODO: Somebody help me write this.
+
+    :param numSVDDims:
+    :return:
+    """
```

Comment:
Note: this should be a private method `_finalizeSVD`.
```
  def finalizeSVD(self, numSVDDims=None):
    """
    Called by finalizeLearning(). This will project all the patterns onto the SVD eigenvectors.
    :param numSVDDims: (int) number of egeinvectors used for projection.
    """
```